server/channels/app/post.go

```diff
@@ -2206,10 +2206,19 @@ func (a *App) GetPostInfo(c request.CTX, postID string) (*model.PostInfo, *model
 			return nil, appErr
 		}
 
-		if team.Type == model.TeamOpen {
-			hasPermissionToAccessTeam = a.HasPermissionToTeam(userID, team.Id, model.PermissionJoinPublicTeams)
-		} else if team.Type == model.TeamInvite {
-			hasPermissionToAccessTeam = a.HasPermissionToTeam(userID, team.Id, model.PermissionJoinPrivateTeams)
+		teamMember, appErr := a.GetTeamMember(channel.TeamId, c.Session().UserId)
+		if appErr == nil {
+			if teamMember.DeleteAt == 0 {
+				hasPermissionToAccessTeam = true
+			}
+		}
```

Comment:
Should we error if `appErr != nil`?
```suggestion
		if appErr != nil {
			return nil, appErr
		}
		if teamMember.DeleteAt == 0 {
			hasPermissionToAccessTeam = true
		}
```

Reply:
No. The logic we want here is the following:
- The user has a team membership?
  - Yes: Is the Memberhship deleted (i.e. has left the team)?
    - Yes: Continue as if it had no membership
    - No: Great, then it has access to the team
  - No: Is the team a team the user can openly join?
    - Yes: Great, then it has access to the team
    - No: Has no permissions to access the team, so return not found

Reply:
But `GetTeamMember` could error because of a reason other than the team membership not existing, right?

Reply:
Sure. We can be more explicit here and check if the error is not found. I will add that.

Reply:
We only want to return early if there is an "unexpected error". So... if there is an error (`appErr != nil`) and it is not a not found error (`appErr.StatusCode != http.StatusNotFound`).

If the membership is not found, we still want to make sure if we have access to the team. So... the current check should be alright.